base/channels.jl

```diff
@@ -414,6 +414,88 @@ function put_unbuffered(c::Channel, v)
     return v
 end
 
+"""
+    append!(c::Channel, iter)
+
+Append all items in `iter` to the channel `c`. If the channel is buffered, this operation requires
+fewer `lock` operations than individual `put!`s. Blocks if the channel is full.
+
+!!! compat "Julia 1.12"
+"""
+function append!(c::Channel{T}, vec::AbstractArray) where {T}
+    # shortcircuit for small vecs
+    if length(vec) == 0
+        return c
+    elseif length(vec) == 1
+        put!(c, @inbounds vec[begin])
```

Comment:
Done

Reply:
I agree on the general point raised here, but hopefully `@inbounds vec[begin]` is fine?  You'd need to have a perverse datatype which defines `firstindex(vec)` to point to an out-of-bound index, which would lead to many other issues.

Reply:
Empty arrays?